platforms/jvm/language-jvm/src/main/java/org/gradle/api/tasks/compile/DebugOptions.java

```diff
@@ -49,35 +49,20 @@ public class DebugOptions extends AbstractOptions {
      *
      * <p>When the value is null, only source and line debugging information will be generated.</p>
      */
-    @Nullable
     @Optional
     @Input
-    @ToBeReplacedByLazyProperty
-    public String getDebugLevel() {
-        return debugLevel;
-    }
+    @ReplacesEagerProperty
+    public abstract Property<String> getDebugLevel();
 
-    /**
-     * Sets which debug information is to be generated during compilation. The value must be a
-     * comma-separated list containing any of the following keywords (without spaces in between):
-     *
-     * <dl>
-     *     <dt>{@code source}
-     *     <dd>Source file debugging information
-     *     <dt>{@code lines}
-     *     <dd>Line number debugging information
-     *     <dt>{@code vars}
-     *     <dd>Local variable debugging information
-     * </dl>
-     *
-     * <p>For example {@code source,lines,vars} is a valid value.</p>
-     *
-     * <p>Alternatively setting the value to {@code none} will disable debug information generation.</p>
-     *
-     * <p>Setting this value to null will reset the property to its default value of only
-     * generating line and source debug information.</p>
-     */
-    public void setDebugLevel(@Nullable String debugLevel) {
-        this.debugLevel = debugLevel;
+    @Override
+    public void define(@Nullable Map<String, Object> args) {
+        // The AbstractOptions.define(Map) expects eager properties but debugOptions is now a lazy property
+        Map<String, Object> mutableArgs = new HashMap<>(args);
```

Comment:
🤔 I wonder if we need this. I see that `CompileOptions` also have bunch of `Property` properties, but we didn't need to add anything like that there

Reply:
I did this to keep `DebugOptionsTest` passing. Because `DebugOptions` is public api I assume this is something (old) builds might depend on.

Reply:
It seems it works without it, or I am doing something wrong?

Reply:
Probably I'm just overly cautious not to break existing builds. I'm fine removing it.